ui/packages/catalog-ui-search/src/main/webapp/js/controllers/ol.layerCollection.controller.js

```diff
@@ -12,197 +12,178 @@
-                }
-            } else if (typeStr === 'WMS') {
-                if (!initObj.params) {
-                    initObj.params = {
-                        LAYERS: initObj.layers
-                    };
-                    _.extend(initObj.params, initObj.parameters);
-                }
-            } else if (typeStr === 'SI') {
-                layerType = ol.layer.Image;
-                if (!initObj.imageExtent) {
-                    initObj.imageExtent = ol.proj.get(properties.projection).getExtent();
-                }
-                if (initObj.parameters) {
-                    _.extend(initObj, initObj.parameters);
-                }
-            } else if (typeStr === 'AGM'){
-                if (initObj.url && initObj.url.indexOf('/tile/{z}/{y}/{x}') === -1) {
-                    initObj.url = initObj.url + '/tile/{z}/{y}/{x}';
-                }
-            } else if (typeStr === 'WMT') {
-                /* If tileMatrixSetID is present (Cesium WMTS keyword) set matrixSet (OpenLayers WMTS keyword) */
-                if (initObj.tileMatrixSetID) {
-                    initObj.matrixSet = initObj.tileMatrixSetID;
-                }
-
-                /* We must mock the tileGrid in order to prevent errors in OpenLayers before the GetCapabilities request returns */
-                initObj.tileGrid = mockTileGrid;
-
-                $.ajax({
-                  url : initObj.url + '?request=GetCapabilities',
-                  success : function(data)  {
-                      var parser = new ol.format.WMTSCapabilities();
-                      var result = parser.read(data);
-                      var options = ol.source.WMTS.optionsFromCapabilities(result, {layer: initObj.layer, matrixSet: initObj.matrixSet});
-                      /* Replace URL with Proxy URL */
-                      options.urls = [initObj.url];
-                      initObj = options;
-                      var layer = new layerType({
-                            visible: model.shouldShowLayer(),
-                            preload: Infinity,
-                            opacity: model.get('alpha'),
-                            source: new type(initObj)
-                        });
-                      var olMapLayers = this.map.getLayers();
-                      olMapLayers.forEach(function(existingLayer, index){
-                        if (existingLayer === this.layerForCid[model.id]){
-                            this.layerForCid[model.id] = layer;
-                            olMapLayers.setAt(index, this.layerForCid[model.id]);
-                        }
-                      }.bind(this));
-                  }.bind(this)
-                });
-            }
+    const res = await window.fetch(url);
+    const text = await res.text();
+    const result = parser.read(text);
+
+    if (result.Contents.Layer.length  === 0) {
+        console.warn(`WMT map layer source ${url} has no layers.`);
```

Comment:
It won't break the UI because https://github.com/codice/ddf/blob/master/ui/packages/ace/lib/webpack.config.js#L196 and it isn't something the user can take action on.

Reply:
I think we should update this to update the layer model and set a warning attribute.  We can then handle when layers have warnings in the layers component in a few different ways.  For example, we can show a warning directly next to the layer.

Reply:
Do you have any advice in terms of what it should look like?